frontend/src/metabase/admin/settings/components/SettingsPages/GeneralSettingsPage.unit.spec.tsx

```diff
@@ -0,0 +1,163 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import {
+  setupDashboardEndpoints,
+  setupPropertiesEndpoints,
+  setupSettingsEndpoints,
+  setupUpdateSettingEndpoint,
+} from "__support__/server-mocks";
+import { renderWithProviders, screen, waitFor } from "__support__/ui";
+import { UndoListing } from "metabase/containers/UndoListing";
+import type { SettingKey } from "metabase-types/api";
+import {
+  createMockDashboard,
+  createMockSettingDefinition,
+  createMockSettings,
+} from "metabase-types/api/mocks";
+
+import { GeneralSettingsPage } from "./GeneralSettingsPage";
+
+const generalSettings = {
+  "site-name": "Metabased",
+  "custom-homepage": true,
+  "custom-homepage-dashboard": 4242,
+  "dismissed-custom-dashboard-toast": false,
+  "site-url": "https://mysite.biz",
+  "admin-email": "help@mysite.biz",
+  "anon-tracking-enabled": false,
+  "redirect-all-requests-to-https": false,
+  "humanization-strategy": "simple",
+  "enable-xrays": false,
+  "allowed-iframe-hosts": "https://cooldashboards.limo",
+} as const;
+
+const setup = async () => {
+  const settings = createMockSettings(generalSettings);
+
+  fetchMock.get("https://mysite.biz/api/health", { status: 200 });
+
+  setupPropertiesEndpoints(settings);
+
+  setupDashboardEndpoints(
+    createMockDashboard({
+      id: 4242,
+      name: "My dashboard",
+    }),
+  );
+
+  setupUpdateSettingEndpoint();
+  setupSettingsEndpoints(
+    Object.entries(settings).map(([key, value]) =>
+      createMockSettingDefinition({ key: key as SettingKey, value }),
+    ),
+  );
+
+  renderWithProviders(
+    <>
+      <GeneralSettingsPage />
+      <UndoListing />
+    </>,
+  );
+
+  await screen.findByText("Redirect to HTTPS");
+};
+
+describe("GeneralSettingsPage", () => {
+  it("should render a GeneralSettingsPage", async () => {
+    await setup();
+
+    [
+      "Site Name",
+      "Site Url",
+      "Redirect to HTTPS",
+      "Custom Homepage",
+      "Email Address for Help Requests",
+      "Anonymous Tracking",
+      "Friendly Table and Field Names",
+      "Enable X-Ray Features",
+      "Allowed domains for iframes in dashboards",
+    ].forEach(text => {
+      expect(screen.getByText(text)).toBeInTheDocument();
+    });
+  });
+
+  it("should make only 4 api calls", async () => {
```

Comment:
Curious about what this test gives us that we don't already have from the tests on other components?

Reply:
this proves that we're not duplicating api calls across different settings components and that caching is working as expected and not multiplying API calls by using the RTK hooks in every component